src/App.tsx

```diff
@@ -15,31 +15,68 @@ import { theme } from "./theme";
 
 library.add(fas, far);
 
-function App() {
-  return (
-    <MuiThemeProvider theme={theme}>
-      <div className={styles.App}>
-        <Header />
-        <div className={styles.app_content_wrapper}>
-          <Router>
-            <Route exact path="/" component={Pages.MainPage} />
-            <Route
-              exact
-              path="/projects/create"
-              component={Pages.CreateProjectPage}
-            />
-            <Route
-              path="/projects/detail/:uuid"
-              component={Pages.ProjectDetailPage}
-            />
-          </Router>
-        </div>
-        <div className={styles.footerPosition}>
-          <Footer />
+// ログインしているユーザーの情報
+type AppStatus = {
+  loginUserFollowingIdList: number[];
+};
+
+type AppProps = {};
+
+class App extends React.Component<AppProps, AppStatus> {
+  constructor(props: AppProps) {
+    super(props);
+    this.state = {
+      loginUserFollowingIdList: [10, 220],
+    };
+  }
+
+  updateFollowingList = (followingIndex: number, userId: number) => {
+    const list = this.state.loginUserFollowingIdList;
+    if (followingIndex === -1) {
```

Comment:
```suggestion
  updateFollowingList = (userId: number) => {
    const list = this.state.loginUserFollowingIdList;
    const followingIndex = list.indexOf(userId);
    if (followingIndex === -1) {
```
適用したあとの変更がちょっとめんどくさいですがこれをすればfollowingIndex引数が必要なくなります

Reply:
本当はなんらかの ID でやりとりするべきなので、インデックスに固めるのはまずいと思います

Reply:
インデックスに固める…というのはどういうことでしょうか?

Reply:
後々 ID を使ったものに変更しづらくなるっていうことです

Reply:
あああなるほど理解しました！！！！！
そしたら今後のためにこのままResolvedにすべきですか？

Reply:
まあフォロー/フォロー解除のボタン押下時の処理にインデックスが必要なのもおかしいですし、
`followingIndex` が `-1` かどうかで動作を変えていたりとなかなか怪しいので、
今のうちに `follow(userIdToFollow)` と `unfollow(userIdToUnfollow)` に分けた方がいいと思います

Reply:
確かに関数分けたほうが良さそうですね

@unios103 だそうです🙇‍♀️